frontend/packages/console-dynamic-plugin-sdk/src/extensions/console-types.ts

```diff
@@ -430,7 +430,14 @@ export type RowReducerFilter<R = any> = RowFilterBase<R> & {
   reducer: (obj: R) => React.ReactText;
 };
 
-export type RowFilter<R = any> = RowMatchFilter<R> | RowReducerFilter<R>;
+export type RowFilter<R = any> = RowMatchFilter<R> | RowReducerFilter<R> | RowSearchFilter<R>;
+
+export type RowSearchFilter<R = any> = {
```

Comment:
I think we can reuse the existing `RowFilterBase` type, just like with other specific filter types.

```ts
export type RowSearchFilter<R = any> = Omit<RowFilterBase<R>, 'items' | 'defaultSelected'> & {
  placeholder?: string;
};
```

Why is the `filter` function optional here? It seems that this function is the core of any filter implementation.

Reply:
Yes right. Fixed